src/transformers/models/mpt/configuration_mpt.py

```diff
@@ -242,35 +245,3 @@ def __init__(
         self.use_cache = use_cache
         self.initializer_range = initializer_range
         super().__init__(**kwargs)
-
-    @property
```

Comment:
OK, maybe cc @ArthurZucker as well as I think he added it through https://github.com/huggingface/transformers/pull/25116. 

After thinking a bit not sure if the issue described in https://github.com/huggingface/transformers/issues/25114 will be resolved after this change. Now it seems that if one passes a dict as `attn_config` it doesn't get converted to a `MptAttentionConfig` and will fail here: https://github.com/huggingface/transformers/blob/main/src/transformers/models/mpt/modeling_mpt.py#L124 , and arthur added it for BC with trust_remote_code behaviour that accepts dict as input for attn_config.

Reply:
I see. I believe this comes from [here](https://github.com/huggingface/transformers/blob/d27e4c18fe2970abcb9a48dcb8a824e48083b15f/src/transformers/configuration_utils.py#L765) and can be fixed at the base config level instead of adding a custom setter. Will push something.